src/js/components/TabPanesComponent.jsx

```diff
@@ -78,9 +78,10 @@ var TabPanesComponent = React.createClass({
   },
 
   updateFilters: function (filters) {
-    var updatedFilters = Object.assign({}, this.state.filters, filters);
-    this.setState({
-      filters: updatedFilters
+    this.setState(function (prevState) {
```

Comment:
I did not know about this, and this makes absolute perfect sense. We should ALWAYS perform this way of doing atomic changes via the callback whenever we need to "update" based on a previous state. This is really good practice!
FTR this is better than `this.setState({filters: Object.assign({}, this.state.filters, filters)}` as it guarantees a "logical" queued execution.